client/src/ClientStack.lua

```diff
@@ -226,33 +230,78 @@ function ClientStack:drawString(string, themePositionOffset, cameFromLegacyScore
   GraphicsUtil.printf(string, x, y, limit, alignment, nil, nil, fontDelta)
 end
 
--- Positions the stack draw position for the given player
-function ClientStack:moveForRenderIndex(renderIndex)
+-- Sets up renderIndex-specific properties and assets
+-- Configures stack positioning parameters for a specific render index (1 for left, 2 for right)
+function ClientStack:setupForRenderIndex(renderIndex)
   self.renderIndex = renderIndex
-  -- Position of elements should ideally be on even coordinates to avoid non pixel alignment
+
   if renderIndex == 1 then
     self.mirror_x = 1
     self.multiplication = 0
   elseif renderIndex == 2 then
     self.mirror_x = -1
     self.multiplication = 1
+  else
+    assert(false)
+    return -- Unknown renderIndex
   end
+  self:assignAssets(GAME.theme:getIngameAssetPack(renderIndex))
+end
+
+-- Calculates the outer edge scaled position for a given render index and offset
+---@param renderIndex integer the render index (1 or 2)
+---@return number outerEdgeScaled the calculated outer edge position
+function ClientStack:calculateOuterEdgeScaled(renderIndex)
   local centerX = (GAME.globalCanvas:getWidth() / 2)
   local stackWidth = self:canvasWidth()
   local innerStackXMovement = 100
   local outerStackXMovement = stackWidth + innerStackXMovement
+  local mirrorX = renderIndex == 1 and 1 or -1
+  
+  return centerX - (outerStackXMovement * mirrorX)
+end
 
-  local outerEdgeScaled = centerX - (outerStackXMovement * self.mirror_x)
+-- Calculates the centered position for a single stack by adjusting standard positioning
+function ClientStack:calculateCenteredStackOuterEdgeScaled()
+  local centerX = (GAME.globalCanvas:getWidth() / 2)
```

Comment:
The function name should indicate that we're calculating for horizontal centering specifically.
I think it would be ideal if the coordinate we center around is passed in as an argument as that would allow the function to be used to place the stack anywhere via center coordinate.